internal/controllers/nmc_reconciler.go

```diff
@@ -749,6 +748,57 @@ func setWorkerContainerArgs(pod *v1.Pod, args []string) error {
 	return nil
 }
 
+func setWorkerSofdepConfig(pod *v1.Pod, modulesLoadingOrder []string) {
+	softdepAnnotationValue := getModulesOrderAnnotationValue(modulesLoadingOrder)
+	setWorkerPodAnnotation(pod, "modules-order", softdepAnnotationValue)
+
+	softdepVolume := v1.Volume{
+		Name: "modules-order",
+		VolumeSource: v1.VolumeSource{
+			DownwardAPI: &v1.DownwardAPIVolumeSource{
+				Items: []v1.DownwardAPIVolumeFile{
+					{
+						Path:     "softdep.conf",
+						FieldRef: &v1.ObjectFieldSelector{FieldPath: "metadata.annotations['modules-order']"},
+					},
+				},
+			},
+		},
+	}
+	softDepVolumeMount := v1.VolumeMount{
+		Name:      "modules-order",
+		ReadOnly:  true,
+		MountPath: "/etc/modprobe.d",
+	}
+	pod.Spec.Volumes = append(pod.Spec.Volumes, softdepVolume)
+	pod.Spec.Containers[0].VolumeMounts = append(pod.Spec.Containers[0].VolumeMounts, softDepVolumeMount)
+}
+
+func getModulesOrderAnnotationValue(modulesNames []string) string {
+	softDepData := ""
```

Comment:
I think we should use a `strings.Builder` here.

Reply:
Fixed